### PR TITLE
Move static helper methods into extensions.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Extracted from https://github.com/dotnet/roslyn/pull/41510 to make that PR easier to review.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.cs [hunk 1]
```diff
@@ -219,5 +219,31 @@ public static bool IsEmbeddedStatementOwner([NotNullWhen(returnValue: true)] thi
                 WhileStatementSyntax n => n.Statement,
                 _ => null,
             };
+
```
Comment (mavasani) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.cs]: The main problem with these small PRs might be they can now end up causing merge conflicts in the refactoring PRs as this file has been moved into shared project. I am not sure though.

File: src/Workspaces/VisualBasic/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.vb [hunk 2]
```diff
@@ -0,0 +1,120 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Runtime.CompilerServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
```
Comment (mavasani) [inline src/Workspaces/VisualBasic/Portable/Extensions/SyntaxNodeExtensions_SharedWithCodeStyle.vb]: This is a new file which can conflict with new file in shared layer which I am adding in the refactoring PR, especially if the code is being merged with another existing extension method file. I feel we are unnecessarily introducing work by creating these small PRs.

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-ide for their views. I think these changes are so simple they can live in the refactoring PR. But if it helps reviewers then I am fine to go this route even if it means each of these can likely cause merge conflicts for the big refactoring PR.

**Comment by mavasani** [comment]
Merging this as it is a very simple refactoring.

**Comment by CyrusNajmabadi** [comment]
> This is a new file which can conflict with new file in shared layer which I am adding in the refactoring PR, 

The overall point I was trying to show here was that we could get these small changes in fast. Then the actual splitting/moving change becomes much smaller in scope as it's only focusing on a couple of things. 

There may be conflicts, yes, with how the existing PR is coming along. But I think that's acceptable because it's the existing PR we want to make much simpler (and this will help do that)

**Comment by mavasani** [comment]
So, if we want to take the path of extracting #41510 into small PRs like these, we have do the following:

1. Create a small PR such as the one created here to extract out smaller pieces of refactoring
2. Handle merge conflicts in the refactoring PR branch when this gets merged. I got merge conflicts in C# code for same helpers added at different place in the file, as the refactoring PR adds many more methods. Additionally, automatic merge resolution ended up adding the new VB extension file that added the extracted helpers, so build fails on the branch has there are now duplicate definitions for these extension methods - one added in this PR into Workspaces layer and one added by the refactoring PR in the shared project.
3. I don't think this simplified the review for the refactoring PR - it excludes the changes to other files in this PR which use the moved extension method, but those changes were extremely trivial anyways. The extension method file move/split are still big enough so they have lesser diffs but nothing to simplify the review.

IMO, this approach of breaking a big refactoring PR into small PRs does not add much value and introduces unnecessary work of extracting code into separate branches and then resolving merge conflicts, and at the end there is still a big refactoring PR to review.

I will still let @dotnet/roslyn-ide make the final call on the preferred approach to aid taking this forward.

**Comment by CyrusNajmabadi** [comment]
> Handle merge conflicts in the refactoring PR branch when this gets merged.

NOrmally this wouldn't happen.  Because the 'full PR' is just built out of the individual smaller PRs.  It's def not easy to go the direction of "full PR split into many small ones".  But if we avoid the 'full PR' approach in the first place, things get much much easier.

**Pull request merged**